### PR TITLE
Fix failing canCompileGLGSExtension browser test

### PR DESCRIPTION
**Pull request by wentsul**
For #457 

Sorry for breaking the browser tests!

#### Change List
- Override userAgent check in `isOldIE` method through `opts` parameter.

#### Notes

Before I noticed that @ibgreen introduced the `opts` param to "inject" the user agent into the `isOldIE` method, I had tried another method that is a bit more complicated but has the benefit of localizing the override as well as not requiring us to pass the test options through the call stack. Code looks like the following:

Add helper method to mock userAgent. This is necessary because `window.navigator` is not writable in the browser environment which is why the tests were only failing when running in the browser.

```
// Helper method to allow us to mock the userAgent across node and browser tests.
function mockUserAgent(value) {
  if (!window.navigator) {
    window.navigator = {};
  }

  const previousUserAgent = window.navigator.userAgent;
  Object.defineProperty(window.navigator, 'userAgent', {
    value,
    writable: true
  });

  return () => {
    mockUserAgent(previousUserAgent);
  };
}
```

In test code:
```
const restoreUserAgent = mockUserAgent(
    'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.162 Safari/537.36');
// call method such as canCompileGLGSExtension();

// reset user agent
restoreUserAgent();
```

If you'd before this style of userAgent mocking, I can update the review. Otherwise, hopefully this resolves the issue for good.

### CODE REVIEW

File: test/src/webgl/context-features.spec.js [hunk 1]
```diff
@@ -78,49 +78,50 @@ test('webgl#caps#hasFeatures(WebGL2)', t => {
...
   const {gl} = fixture;
-  const oldNavigator = window.navigator;
+
+  const userAgentNonIE = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.162 Safari/537.36';
+  const userAgentOldIE = 'Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko';
 
   t.ok(typeof canCompileGLGSExtension === 'function', 'canCompileGLGSExtension defined');
 
-  // Non-IE version.
+  // // Non-IE version.
```
Comment (ibgreen) [inline test/src/webgl/context-features.spec.js]: Double comment?

File: test/src/webgl/context-features.spec.js [hunk 2]
```diff
@@ -78,49 +78,50 @@ test('webgl#caps#hasFeatures(WebGL2)', t => {
...
   const getShaderParameterStub = sinon.stub(gl, 'getShaderParameter');
```
Comment (ibgreen) [inline test/src/webgl/context-features.spec.js]: Add comment explain why we are checking whether this particular function was called?

File: test/src/webgl/context-features.spec.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
-import {hasFeature, hasFeatures, getFeatures, FEATURES} from 'luma.gl';
+import {canCompileGLGSExtension, hasFeature, hasFeatures, getFeatures, FEATURES} from 'luma.gl';
 import test from 'tape-catch';
-// import sinon from 'sinon';
+import sinon from 'sinon';
```
Comment (ibgreen) [inline test/src/webgl/context-features.spec.js]: I also want to get rid of the sinon dependency. Opened https://github.com/uber-web/probe.gl/pull/24 as first step.

Comment (wentsul) [reply test/src/webgl/context-features.spec.js]: Sounds good- once that's available, I'll update this PR.

File: test/src/webgl/context-features.spec.js (hunk 2 above)
Comment (wentsul) [reply test/src/webgl/context-features.spec.js]: Updated the test assertion comments to be more descriptive.

**Comment by wentsul** [comment]
Updated tests to use `probe.gl` instead of `sinon`. I couldn't complete remove `sinon` yet due to the tests written in `assemble-shaders.spec.js`. They use some advanced stubbing to confirm call params.

**Pull request merged**